src/main/java/com/google/devtools/build/lib/remote/RemoteRetrier.java

```diff
@@ -16,43 +16,21 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
 import com.google.common.util.concurrent.ListeningScheduledExecutorService;
 import io.grpc.Status;
 import io.grpc.StatusException;
 import io.grpc.StatusRuntimeException;
+import java.io.IOException;
 import java.time.Duration;
 import java.util.concurrent.Callable;
 import java.util.concurrent.ThreadLocalRandom;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 
-/**
- * Specific retry logic for remote execution/caching.
- *
- * <p>A call can disable retries by throwing a {@link PassThroughException}. <code>
- *   RemoteRetrier r = ...;
- *   try {
- *    r.execute(() -> {
- *      // Not retried.
- *      throw PassThroughException(new IOException("fail"));
- *    }
- *   } catch (RetryException e) {
- *     // e.getCause() is the IOException
- *     System.out.println(e.getCause());
- *   }
- * </code>
- */
+/** Specific retry logic for remote execution/caching. */
 public class RemoteRetrier extends Retrier {
```

Comment:
I think we can kill the `RemoteRetrier` class and just use `Retrier`?

Reply:
We maybe could. It depends how much you want to keep `Retrier` generic. Even after this change, `RemoteRetrier.execute` encodes that application-specific logic that `IOException` and `InterruptedException` should be directly propagated while other exceptions should be wrapped in a `RuntimeException`.

There are several cleanups including removing `RemoteRetrier` we could make after this change. As you mention, this CL is already quite complex, so I'm reluctant to stick them in here.

Reply:
sgtm